core/src/main/scala/kafka/log/Log.scala

```diff
@@ -387,7 +387,18 @@ class Log(@volatile var dir: File,
         fileSuffix = SwapFileSuffix)
       info(s"Found log file ${swapFile.getPath} from interrupted swap operation, repairing.")
       recoverSegment(swapSegment)
-      val oldSegments = logSegments(swapSegment.baseOffset, swapSegment.readNextOffset)
+
+      // Assigned to a new value
+      val nextOffset = swapSegment.readNextOffset
+      var oldSegments = logSegments(swapSegment.baseOffset, nextOffset)
```

Comment:
If the goal of this is to delete .log file if .swap file coexists, why not delete it at the removeTempFilesAndCollectSwapFiles?

Reply:
We could probably.. We have to investigate. Lets do this way and we can re-revisit it later